lib/gc/Transforms/IterativeTilingAndFusion.cpp

```diff
@@ -256,11 +274,48 @@ SingleCandidateInBlockFilter(RewriterBase &rewriter,
         FailureOr<SmallVector<OpOperand *>> realConsumers =
             scfX::getRealConsumersFromInsertSliceOp(otherCandidate,
                                                     forwardSlice);
+        // Record operand pointer of same owner.
+        OpOperand *operandOfSameOwner = nullptr;
         if (succeeded(realConsumers) &&
-            llvm::any_of(*realConsumers, [&defOrUse](OpOperand *use) {
-              return use->getOwner() == defOrUse.ownerOp;
-            }))
-          return failure();
+            llvm::any_of(*realConsumers,
+                         [&defOrUse, &operandOfSameOwner](OpOperand *use) {
+                           if (use->getOwner() != defOrUse.ownerOp)
+                             return false;
+                           operandOfSameOwner = use;
+                           return true;
+                         })) {
+          assert(operandOfSameOwner);
```

Comment:
It is better to put the error message in the assert following https://llvm.org/docs/CodingStandards.html#assert-liberally

Reply:
Ok, thanks for reminder. Added.